apps/cowswap-frontend/src/common/pure/OrderProgressBarV2/index.tsx

```diff
@@ -355,79 +495,101 @@ function FinishedStep({
-              theme="light"
-              overrideColor={UI.COLOR_PRIMARY_DARKER}
-              height={19}
-            />
-          </styledEl.FinishedLogo>
-          <styledEl.FinishedTagLine>
-            {showSurplus ? (
-              <>
-                ...gets you <b>moooooore.</b>
-              </>
-            ) : (
-              <>
-                Did you <b>know?</b>
-              </>
-            )}
-          </styledEl.FinishedTagLine>
+        <styledEl.ProgressImageWrapper bgColor={'#65D9FF'} padding={'10px'} gap={'10px'}>
           <styledEl.CowImage>
-            <SVG src={PROGRESSBAR_COW_SURPLUS} />
+            <styledEl.ShareButton onClick={shareOnTwitter}>
+              <SVG src={ICON_SOCIAL_X} />
+              <span>Share this {shouldShowSurplus ? 'win' : 'tip'}!</span>
+            </styledEl.ShareButton>
+            <SVG
+              src={React.useMemo(() => {
+                const images = [
+                  PROGRESSBAR_COW_SURPLUS_1,
+                  PROGRESSBAR_COW_SURPLUS_2,
+                  PROGRESSBAR_COW_SURPLUS_3,
+                  PROGRESSBAR_COW_SURPLUS_4,
+                ]
+                return images[Math.floor(Math.random() * images.length)]
+              }, [])}
+            />
           </styledEl.CowImage>
-          {showSurplus && (
-            <>
-              <styledEl.TokenPairTitle>
-                <span>Swap order</span>
-                <b>{order ? `${order.inputToken.symbol}/${order.outputToken.symbol}` : 'N/A'}</b>
-              </styledEl.TokenPairTitle>
-              <styledEl.TokenImages>
-                <TokenLogo token={order?.inputToken} size={34} />
-                <TokenLogo token={order?.outputToken} size={34} />
-              </styledEl.TokenImages>
-            </>
-          )}
-          <styledEl.Surplus showSurplus={!!showSurplus}>
-            {showSurplus ? (
-              <>
-                <span>Your surplus</span>
-                {surplusPercent ? <b>+{parseFloat(surplusPercent).toFixed(2)}%</b> : <b>N/A</b>}
-              </>
-            ) : (
-              <>
-                <span>Unlike other exchanges, here you don't pay any fees if your trade fails.</span>
-              </>
-            )}
-          </styledEl.Surplus>
+          <styledEl.FinishedImageContent>
+            <styledEl.FinishedTagLine>
+              {shouldShowSurplus ? (
+                <styledEl.BenefitSurplusContainer ref={surplusContainerRef}>
+                  <span ref={surplusTextRef} style={{ fontSize: `${surplusFontSize}px` }}>
+                    I just received surplus on my
+                    <styledEl.TokenPairTitle
+                      title={order ? `${order.inputToken.symbol}/${order.outputToken.symbol}` : 'N/A'}
+                    >
+                      {order
+                        ? truncateWithEllipsis(`${order.inputToken.symbol}/${order.outputToken.symbol}`, 30)
+                        : 'N/A'}
```

Comment:
I rather move the `order` check up and not display anything when it's not set.
Unlikely to happen, and makes the code cleaner.

Reply:
Addressed this one now.